### PR TITLE
Cypress/fe 1640

### PR DESCRIPTION
**Pull request by DlgSHi**
# automated BDD test specs for:
- SettingsRow, 
- ShowEditPod, 
- Sidebar, 
- SimpleColorPicker;

### CODE REVIEW

File: cypress/integration/showEditPod.feature [hunk 1]
```diff
@@ -0,0 +1,155 @@
+Feature: Show Edit Pod component
+  I want to test Show Edit Pod component
+
+  Background: Open Show Edit Pod component page
+    Given I open "ShowEditPod" component page
+
+ @positive
+  Scenario: Check the content of the component
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/locators/show-edit-pod/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import {
+  SHOW_EDIT_POD_PREVIEW, SHOW_EDIT_POD_TITLE, SHOW_EDIT_POD_COLLAPSIBLE_CONTENT,
+  SHOW_EDIT_SAVE_BUTTON, SHOW_EDIT_CANCEL_BUTTON, SHOW_EDIT_DELETE_BUTTON,
+  SHOW_EDIT_POD_EDIT, SHOW_EDIT_POD_SECONDARY_BLOCK, SHOW_EDIT_POD_SECONDARY_WRAPPER,
+} from './locators';
+
+// component preview locators
+export const showEditPodPreview = () => cy.iFrame(SHOW_EDIT_POD_PREVIEW);
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/locators/show-edit-pod/index.js]: fixed

File: cypress/integration/showEditPod.feature [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+ @positive
+  Scenario: Check the content of the component 
+    # When I open "ShowEditPod" component page
+    Then Show Edit Pod component has proper content inside itself
+
+  @positive
+  Scenario Outline: Change Show Edit Pod cancelText to <cancelText>
+    When I set cancelText to "<cancelText>"
+      And I click onto edit button
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/showEditPod.feature [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    When I select buttonAlign to "<buttonAlign>"
+      And I click onto edit button
+    Then Show Edit Pod buttonAlign value is set to "<buttonAlign>"
+    Examples:
+      | buttonAlign |
+      | right       |
+      | left        |    
+
+  @positive
+  Scenario: Check the delete click event
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/showEditPod.feature [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    When I click onto edit button
+      And clear all events in Actions Tab
+      And I click delete button
+    Then delete function was called in Actions Tab
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: agreed, fixed

File: cypress/integration/showEditPod.feature [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+  @positive
+  Scenario: Check the cancel click event
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/showEditPod.feature [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    When I click onto edit button
+      And clear all events in Actions Tab
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/showEditPod.feature [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+      And I click cancel button
+    Then cancel function was called in Actions Tab
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/sidebar.feature [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      | medium       | 450              |
+      | medium-large | 550              |
+      | large        | 650              |
+      | extra-large  | 750              |
+
+  @positive
+  Scenario: Check the cancel click event
+    When I check open checkbox
+      And clear all events in Actions Tab
+      And I click onto cross icon
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/sidebar.feature]: fixed

File: cypress/integration/sidebar.feature [hunk 10]
```diff
@@ -0,0 +1,61 @@
+Feature: Sidebar component
+  I want to test Sidebar component
+
+  Background: Open Sidebar component page
+    Given I open "Sidebar" component page
+
+  @positive
+  Scenario: Enable open checkbox for a Sidebar component
+    When I check open checkbox
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/sidebar.feature]: fixed

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
+import { simpleColorPickerInput, simpleColorPickerDiv, simpleColorPickerPreview } from '../../locators/simple-color-picker';
+
+const ZERO_ELEMENT = 0;
+const FIRST_ELEMENT = 1;
+const SECOND_ELEMENT = 2;
+const THIRD_ELEMENT = 3;
+
+Then('Simple Color Picker name on preview is set to {string}', (value) => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: it's an additional assertion, we could verify that the correct hex code was accepted.

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  simpleColorPickerInput(FIRST_ELEMENT)
+    .should('have.attr', 'name', `${value}`)
+    .and('have.css', 'position', 'absolute')
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: deleted

File: cypress/support/step-definitions/sidebar-steps.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+    .and('have.css', 'opacity', '0.6')
+    .and('have.css', 'position', 'fixed')
+    .and('have.css', 'right', '0px')
+    .and('have.css', 'top', '0px');
+});
+
+Then('Sidebar component is visible', () => {
+  sidebarPreview().should('be.visible');
+  sidebarPreview()
+    .should('have.css', 'border-radius', '1px')
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/sidebar-steps.js]: fixed

File: cypress/support/step-definitions/sidebar-steps.js [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+    .and('have.css', 'bottom', '0px')
+    .and('have.css', 'top', '0px')
+    .and('have.css', 'z-index', '1002')
+    .and('have.css', 'background-color', 'rgb(230, 235, 237)')
+    .and('have.css', 'padding', '20px')
+    .and('have.css', 'border-left', '1px solid rgb(204, 214, 219)')
+    .and('have.css', 'box-shadow', 'rgba(0, 0, 0, 0.05) -10px 0px 15px 0px')
+    .and('have.css', 'right', '0px');
+  sidebarHeader()
+    .should('have.css', 'background-color', 'rgb(255, 255, 255)')
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/sidebar-steps.js]: deleted

File: cypress/integration/simpleColorPicker.feature [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    Examples:
+      | availableColors         | length |
+      | #AAAAAA                 | 1      |
+      | #C8DCFC,#FFFF00         | 2      | 
+      | #AAAAAA,#255BC7,#ED1C5F | 3      | 
+
+  @positive
+  Scenario Outline: Check the Simple Color Picker <position> element was selected
+    When clear all events in Actions Tab
+      And I click onto "<position>" element
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/simpleColorPicker.feature]: I don't think it is a good idea, coz we need to check does the first / second / third element was selected.

File: cypress/support/step-definitions/settings-row-steps.js [hunk 16]
```diff
@@ -0,0 +1,37 @@
+import {
+  settingsRowTitle, settingsRowChildren, settingsRowDescription,
+  settingsRowPreview,
+} from '../../locators/settings-row';
+
+const DIVIDER_CLASS = 'carbon-settings-row--has-divider';
+
+Then('Settings Row {word} on preview is set to {string}', (parameter, text) => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/settings-row-steps.js]: agreed, divided

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+    .and('have.css', 'width', '56px');
+  simpleColorPickerInput(THIRD_ELEMENT)
+    .should('have.attr', 'name', `${value}`)
+    .and('have.css', 'position', 'absolute')
+    .and('have.css', 'opacity', '0')
+    .and('have.css', 'height', '56px')
+    .and('have.css', 'width', '56px');
+});
+
+Then('Simple Color Picker availableColors on preview is set to {string} with rgb {string} parameter', (color, rgb) => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: had a talk with U about this :)

File: cypress/integration/simpleColorPicker.feature [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    Then Simple Color Picker availableColors on preview is set to "<availableColors>" with rgb "<rgb>" parameter
+    Examples:
+      | availableColors             | rgb                                                                   |
+      | #AAAAAA,#255BC719,#ED1C5F   | rgb(170, 170, 170);rgba(37, 91, 199, 0.098);rgb(237, 28, 95)          |
+      | #C8DCFC4C,#FFFF00,#92007699 | rgba(200, 220, 252, 0.298);rgb(255, 255, 0);rgba(146, 0, 118, 0.6)    |
+      | #CAC1DC59,#067B5BE6,#BAFA61 | rgba(202, 193, 220, 0.35);rgba(6, 123, 91, 0.9);rgb(186, 250, 97)     |
+      | #7DBE27,#273B0C,#00FF0080   | rgb(125, 190, 39);rgb(39, 59, 12);rgba(0, 255, 0, 0.5)                |
+
+  @positive
+  Scenario Outline: Change Simple Color Picker availableColors to <availableColors> and check the length of elements
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/simpleColorPicker.feature]: due to not working functionality for Opacity nad RGBA no need to add those kind of tests

File: cypress/support/step-definitions/show-edit-pod-steps.js [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+const SHOW_EDIT_POD_NO_BORDER_CONTENT = 'carbon-pod--no-border';
+const SHOW_EDIT_POD_WRAPPER = 'carbon-form__buttons--';
+const FIRST_DIV = 1;
+const SECOND_DIV = 2;
+const THIRD_DIV = 3;
+const INNER_CONTENT_TITLE = 'title';
+const INNER_CONTENT_BODY = 'body';
+
+
+Then('Show Edit Pod {word} on preview is set to {string}', (parameter, text) => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/show-edit-pod-steps.js]: done

File: cypress/support/step-definitions/show-edit-pod-steps.js [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+Then('Show Edit Pod component has no saving property and should be saved manualy', () => {
+  showEditPodSaveButton()
+    .should('not.have.attr', 'disabled');
+  showEditPodSaveButton()
+    .should('have.css', 'background', 'rgb(37, 91, 199) none repeat scroll 0% 0% / auto padding-box border-box')
+    .and('have.css', 'border-color', 'rgba(0, 0, 0, 0)');
+});
+
+Then('Show Edit Pod {word} value is set to {string}', (word, property) => {
+  switch (word) {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/show-edit-pod-steps.js]: done

File: cypress/support/step-definitions/sidebar-steps.js [hunk 21]
```diff
@@ -0,0 +1,62 @@
+import { sidebarPreview, sidebarHeader } from '../../locators/sidebar';
+import { backgroundUILocator } from '../../locators';
+
+Then('Sidebar {word} value is set to {string}', (parameter, value) => {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/sidebar-steps.js]: done

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 22]
```diff
@@ -69,39 +31,19 @@ Then('Simple Color Picker availableColors on preview is set to {string}', (value
...
-      break;
-    case 'second':
-      simpleColorPickerInput(SECOND_ELEMENT).click();
-      break;
-    case 'third':
-      simpleColorPickerInput(THIRD_ELEMENT).click();
-      break;
-    default: throw new Error('There is no such index of element');
+When('I click onto {int} of element', (index) => {
+  for (let i = 1; i < index + 1; ++i) {
```
Comment (marcinstachowicz) [inline cypress/support/step-definitions/simple-color-picker-steps.js]: iterate from 0 to i<index and make +1 here: `simpleColorPickerInput(i+1).click();`

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 23]
```diff
@@ -69,39 +31,19 @@ Then('Simple Color Picker availableColors on preview is set to {string}', (value
...
-      simpleColorPickerDiv(THIRD_ELEMENT).children()
-        .should('have.attr', 'data-element', 'tick')
-        .should('have.class', 'icon-tick');
-      break;
-    default: throw new Error('There is no such index of element');
+  if (index === '1') {
+    simpleColorPickerInput(SECOND_ELEMENT).click();
+    simpleColorPickerInput(FIRST_ELEMENT).click();
+  }
+  for (let i = 1; i < index; i++) {
```
Comment (marcinstachowicz) [inline cypress/support/step-definitions/simple-color-picker-steps.js]: please double check that loop

File: cypress/integration/showEditPod.feature [hunk 24]
```diff
@@ -42,7 +42,7 @@ Feature: Show Edit Pod component
   @positive
   Scenario Outline: Change Show Edit Pod saveText to <saveText>
     When I set saveText to "<saveText>"
-      And I click onto edit button
+      And I edit component
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/showEditPod.feature]: fixed

File: cypress/integration/simpleColorPicker.feature [hunk 25]
```diff
@@ -23,11 +23,12 @@ Feature: Simple Color Picker component
     When I set availableColors to "<availableColors>"
     Then Simple Color Picker availableColors on preview is set to "<availableColors>" with rgb "<rgb>" parameter
     Examples:
-      | availableColors             | rgb                                                                   |
-      | #AAAAAA,#255BC719,#ED1C5F   | rgb(170, 170, 170);rgba(37, 91, 199, 0.098);rgb(237, 28, 95)          |
-      | #C8DCFC4C,#FFFF00,#92007699 | rgba(200, 220, 252, 0.298);rgb(255, 255, 0);rgba(146, 0, 118, 0.6)    |
-      | #CAC1DC59,#067B5BE6,#BAFA61 | rgba(202, 193, 220, 0.35);rgba(6, 123, 91, 0.9);rgb(186, 250, 97)     |
-      | #7DBE27,#273B0C,#00FF0080   | rgb(125, 190, 39);rgb(39, 59, 12);rgba(0, 255, 0, 0.5)                |
+      | availableColors               | rgb                                                                        |
+      # comented, this functionality doesn't work on Electron / IE
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/simpleColorPicker.feature]: FE-1652 - reported

File: cypress/support/step-definitions/settings-row-steps.js [hunk 26]
```diff
@@ -5,23 +5,19 @@ import {
...
-      break;
-    case 'description':
-      settingsRowDescription()
-        .should('have.text', `${text}`);
-      break;
-    default:
-      throw new Error('Not a Settings Row component input property');
-  }
+Then('Settings Row title on preview is set to {string}', (text) => {
+  settingsRowTitle()
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/settings-row-steps.js]: no :), fixed

File: cypress/integration/sidebar.feature [hunk 27]
```diff
@@ -55,7 +51,7 @@ Feature: Sidebar component
 
   @positive
   Scenario: Check the cancel click event
-    When I check open checkbox
-      And clear all events in Actions Tab
-      And I click onto cross icon
-    Then cancel function was called in Actions Tab
\ No newline at end of file
+    # When I check open checkbox
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/sidebar.feature]: fixed

File: cypress/integration/simpleColorPicker.feature [hunk 28]
```diff
@@ -36,17 +37,17 @@ Feature: Simple Color Picker component
...
-      | #AAAAAA,#255BC7,#ED1C5F | 3      | 
+      | #C8DCFC,#FFFF00         | 2      |
+      | #AAAAAA,#255BC7,#ED1C5F | 3      |
 
   @positive
   Scenario Outline: Check the Simple Color Picker <position> element was selected
-    When clear all events in Actions Tab
-      And I click onto "<position>" element
+    When clear all actions in Actions Tab
+      And I click onto <position> of element
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/simpleColorPicker.feature]: fixed

File: cypress/support/step-definitions/simple-color-picker-steps.js (hunk 22 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: fixed

File: cypress/support/step-definitions/simple-color-picker-steps.js (hunk 23 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: checked, U was right :) fixed

File: cypress/support/step-definitions/simple-color-picker-steps.js [hunk 29]
```diff
@@ -69,39 +31,19 @@ Then('Simple Color Picker availableColors on preview is set to {string}', (value
...
-        .should('have.attr', 'data-element', 'tick')
-        .should('have.class', 'icon-tick');
-      break;
-    case 'third':
-      simpleColorPickerDiv(THIRD_ELEMENT).children()
-        .should('have.attr', 'data-element', 'tick')
-        .should('have.class', 'icon-tick');
-      break;
-    default: throw new Error('There is no such index of element');
+  if (index === '1') {
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/support/step-definitions/simple-color-picker-steps.js]: moved

**Pull request merged**